### PR TITLE
"merrittlj" (Keebio/Iris) keymap

### PR DESCRIPTION
**Pull request by merrittlj**
Add my specialized Dvorak-based Iris keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by merrittlj** [comment]
Sorry, accidentally opened it early.

File: keyboards/keebio/iris/keymaps/merrittlj/keymap.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 Danny Nguyen (@nooges)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+#define _DVORAK 0
+#define _SHIFT 1
+#define _SYMBOL 2
+#define _FUNCTION 3
+#define _ADDITIONAL 4
```
Comment (dunk2k) [inline keyboards/keebio/iris/keymaps/merrittlj/keymap.c:10]:
```suggestion
enum layers {
    _DVORAK,
    _SHIFT,
    _SYMBOL,
    _FUNCTION,
    _ADDITIONAL,
};
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/keebio/iris/keymaps/merrittlj/keymap.c [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+#define KC_BANG S(KC_1)
+#define KC_AT S(KC_2)
+#define KC_HASH S(KC_3)
+// #define KC_DOLLAR S(KC_4)
+#define KC_PCNT S(KC_5)
+#define KC_AHEAD S(KC_6)
+#define KC_AND S(KC_7)
+// #define KC_ASTERISK S(KC_8)
+#define KC_LPAREN S(KC_9)
+#define KC_RPAREN S(KC_0)
```
Comment (dunk2k) [inline keyboards/keebio/iris/keymaps/merrittlj/keymap.c:32]: optimisation opportunity by utilising [shifted keycodes](https://docs.qmk.fm/#/keycodes_us_ansi_shifted).

File: keyboards/keebio/iris/keymaps/merrittlj/keymap.c (hunk 1 above)
Comment (merrittlj) [reply keyboards/keebio/iris/keymaps/merrittlj/keymap.c:10]: I thought I had something like that.. I will fix that.

File: keyboards/keebio/iris/keymaps/merrittlj/keymap.c (hunk 2 above)
Comment (merrittlj) [reply keyboards/keebio/iris/keymaps/merrittlj/keymap.c:32]: I didn't see the shifted keycodes on the all-keycodes page, let me fix all of them.

**Comment by merrittlj** [comment]
@dunk2k Is the code up-to-date with what you suggested?

File: keyboards/keebio/iris/keymaps/merrittlj/README.MD [hunk 3]
```diff
@@ -0,0 +1,15 @@
+![merrittlj keymap images](https://imgur.com/a/7AL7Yxq)
```
Comment (dunk2k) [inline keyboards/keebio/iris/keymaps/merrittlj/README.MD:1]: Unfortunately you'll have to get a link for every image in this link, including the added `h`, and add them in sequence.

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
> @dunk2k Is the code now properly up-to-date with what you suggested?

@merrittlj utilising the shifted keycodes in the same manner as you intended 👍

File: keyboards/keebio/iris/keymaps/merrittlj/README.MD (hunk 3 above)
Comment (merrittlj) [reply keyboards/keebio/iris/keymaps/merrittlj/README.MD:1]: Is stitching all of the layer images together into one image, an option?

Comment (dunk2k) [reply keyboards/keebio/iris/keymaps/merrittlj/README.MD:1]: > What do you mean exactly by "the added h"?
Taken from PR checklist:
- a picture about the keyboard and preferably about the PCB, too
    - images are not to be placed in the qmk_firmware repository
    - images should be uploaded to an external image hosting service, such as imgur.
    - if imgur is used, images should be resized appropriately: append “h” to the image url i.e. https://i.imgur.com/vqgE7Ok.jpg becomes https://i.imgur.com/vqgE7Okh.jpg
    - image links should link directly to the image, not a “preview” – i.e. https://imgur.com/vqgE7Ok should be https://i.imgur.com/vqgE7Okh.jpg when using imgur

Comment (dunk2k) [reply keyboards/keebio/iris/keymaps/merrittlj/README.MD:1]: > Is stitching all of the layer images together into one image, an option?

Not from imgur directly, you'd need to stitch them together and upload the result to imgur.

**Comment by merrittlj** [comment]
The link is now corrected. However, I am seeing that the builds are failing due to the firmware size. I'm certain a larger size is due to the RGB custom animations, and the extra QMK-specific. This was compiling successfully for my keyboard(rev5), with enough space. Shall I remove some of the extra components to shrink the size, or are these errors okay?

**Comment by dunk2k** [comment]
Could try adding recommendations from [squeezing the most out of avr](https://docs.qmk.fm/#/squeezing_avr?id=squeezing-the-most-out-of-avr)

**Comment by merrittlj** [comment]
All builds succeeded, ready to merge now.

**Comment by merrittlj** [comment]
Please hold, I updated the symbol layer and want to update this PR.

**Comment by merrittlj** [comment]
I have fully finished now, feel free to merge whenever.

File: keyboards/keebio/iris/keymaps/merrittlj/README.MD
Comment (drashna) [inline keyboards/keebio/iris/keymaps/merrittlj/README.MD:1]: please rename this file to be all lower case.

**Comment by merrittlj** [comment]
@drashna The changes have been implemented, but GitHub is still showing me that the changes are still requested(I resolved them after implementation). Is this normal? The PR should be ultimately ready to merge.

File: keyboards/keebio/iris/keymaps/merrittlj/keymap.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case DVORAK:
+        if (record->event.pressed) {
+            set_single_persistent_default_layer(_DVORAK);
+        }
+        return false;
+        break;
+    case SHIFT:
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/merrittlj/keymap.c:102]: all of these can be replaced with just `MO(x)` keycodes.

**Comment by merrittlj** [comment]
@drashna The changes you requested have been implemented, as well as key overrides and the removal of the custom shift layer. However, even after the removal of several RGB animations, the firmware size is still large for rev. 6 and rev. 7 for the Iris keyboard. Is this acceptable to merge? Is it up to the user to further optimize if they have the affected boards, or should I optimize this keymap on my own(with the loss of some features/quality)?

**Pull request closed without merging**